### ISSUE TITLE
sockets-tests-client/server need love (or the axe)

### ISSUE THREAD

**Issue by cgay** [dylan-lang/opendylan#1102]
Need to assess the utility of these two supposed tests and either fix them or delete them.

```shell
$ dylan-compiler -build network/tests/client/sockets-tests-client.lid
$ sources/_build/bin/sockets-tests-client
Running suite sockets-test-suite:
Running test sockets-tests: CRASHED in 0.001791s and 0B
    socket error 111
Completed suite sockets-test-suite: FAILED in 0.001791s
```
```shell
$ dylan-compiler -build network/tests/server/sockets-tests-server.lid
$ _build/bin/sockets-tests-server
socket error 13
socket error 13
Breaking into debugger.
Trace/breakpoint trap
```

**Comment by pedro-w** [comment]
I had a look at this. 
* the client is quite fragile and crashes if the server isn't running (that's error 111)
* the server uses privileged ports (7 & 13) hence you'd have to run it with `sudo` (error 13)
* It tries to get the remote port of a (UDP) socket that's not connected, but there's no error handling on Unix, hence it crashes instead of reporting not connected. See below:
https://github.com/dylan-lang/opendylan/blob/e0b6fa759a98b2904672156657850e54c1121642/sources/network/sockets/unix-socket-accessor.dylan#L524-L531
(in general the Unix code looks unfinished, like the Windows stuff was copied across and anything winsock-specific just commented out)

A bigger problem is that it uses C-FFI but the mapping to C types isn't always right. It probably was  OK for 32-bit Dylan. See for example:
https://github.com/dylan-lang/opendylan/blob/e0b6fa759a98b2904672156657850e54c1121642/sources/network/sockets/unix-socket-accessor.dylan#L410-L413
The size of the numeric address for AF_INET (second parameter to `gethostbyaddr`) should be 4, not `size-of(<C-raw-unsigned-long>)` which is 8 on 64-bit Linux/macOS.
Strangely this causes problems for Linux but not macOS, further down the line.

So, we will need to go through the network library quite carefully to track all these down.

Finally, the  actual sockets-tests-client/server applications are somewhere between a test and an example. It's not possible to run them as part of `make check` because the server needs starting before the client runs. However it should be straightforward to produce a combined test app which will communicate with itself via sockets and I think this would be useful.

I've started a branch to see what I can bodge up to make it work; at the moment the tests pass on mac but not linux due to the size issue I mentioned above.
https://github.com/pedro-w/opendylan/tree/bug-1102

TL;DR - Love, not the axe (but tough love ..!)
